ibm/config.go

```diff
@@ -594,91 +594,368 @@ func (sess clientSession) IAMNamespaceAPI() (*ns.IbmCloudFunctionsNamespaceAPIV1
+			Authenticator: sess.cisPageRuleClient.Service.Options.Authenticator,
+		}
+		cisPageRuleClient, cisPageRuleErr := cispagerulev1.NewPageRuleApiV1(cisPageRuleOpt)
+		if cisPageRuleErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while cofiguring CIS Page Rule client: %s",
+				cisPageRuleErr)
+		}
+		return cisPageRuleClient, nil
+	}
 	return sess.cisPageRuleClient, sess.cisPageRuleErr
 }
 
-// cCIS Edge Function
+// CIS Edge Function
 func (sess clientSession) CisEdgeFunctionClientSession() (*cisedgefunctionv1.EdgeFunctionsApiV1, error) {
+	if sess.cisEdgeFunctionErr == nil {
+		cisEdgeFunctionOpt := &cisedgefunctionv1.EdgeFunctionsApiV1Options{
+			URL:            sess.cisEdgeFunctionClient.Service.Options.URL,
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisEdgeFunctionClient.Service.Options.Authenticator,
+		}
+		cisEdgeFunctionClient, cisEdgeFunctionErr :=
+			cisedgefunctionv1.NewEdgeFunctionsApiV1(cisEdgeFunctionOpt)
+		if cisEdgeFunctionErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS Edge Function client: %s",
+				cisEdgeFunctionErr)
+		}
+		return cisEdgeFunctionClient, nil
+	}
 	return sess.cisEdgeFunctionClient, sess.cisEdgeFunctionErr
 }
 
 // CIS SSL certificate
 func (sess clientSession) CisSSLClientSession() (*cissslv1.SslCertificateApiV1, error) {
+	if sess.cisSSLErr == nil {
+		cisSSLOpt := &cissslv1.SslCertificateApiV1Options{
+			URL:            sess.cisSSLClient.Service.Options.URL,
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisSSLClient.Service.Options.Authenticator,
+		}
+
+		cisSSLClient, cisSSLErr := cissslv1.NewSslCertificateApiV1(cisSSLOpt)
+		if cisSSLErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS SSL certificate client: %s",
+				cisSSLErr)
+		}
+		return cisSSLClient, nil
+	}
 	return sess.cisSSLClient, sess.cisSSLErr
 }
 
 // CIS WAF Packages
 func (sess clientSession) CisWAFPackageClientSession() (*ciswafpackagev1.WafRulePackagesApiV1, error) {
+	if sess.cisWAFPackageErr == nil {
+		cisWAFPackageOpt := &ciswafpackagev1.WafRulePackagesApiV1Options{
+			URL:           sess.cisWAFPackageClient.Service.Options.URL,
+			Crn:           core.StringPtr(""),
+			ZoneID:        core.StringPtr(""),
+			Authenticator: sess.cisWAFPackageClient.Service.Options.Authenticator,
+		}
+		cisWAFPackageClient, cisWAFPackageErr :=
+			ciswafpackagev1.NewWafRulePackagesApiV1(cisWAFPackageOpt)
+		if cisWAFPackageErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuration CIS WAF Package client: %s",
+				cisWAFPackageErr)
+		}
+		return cisWAFPackageClient, nil
+	}
 	return sess.cisWAFPackageClient, sess.cisWAFPackageErr
 }
 
 // CIS Zone Settings
 func (sess clientSession) CisDomainSettingsClientSession() (*cisdomainsettingsv1.ZonesSettingsV1, error) {
+	if sess.cisDomainSettingsErr == nil {
```

Comment:
Same here:

```go
if sess.cisDomainSettingsErr != nil {
	return sess.cisDomainSettingsClient, sess.cisDomainSettingsErr
}

cisDomainSettingsOpt := &cisdomainsettingsv1.ZonesSettingsV1Options{
	URL:            sess.cisDomainSettingsClient.Service.Options.URL,
	Crn:            core.StringPtr(""),
	ZoneIdentifier: core.StringPtr(""),
	Authenticator:  sess.cisDomainSettingsClient.Service.Options.Authenticator,
}
cisDomainSettingsClient, cisDomainSettingsErr :=
	cisdomainsettingsv1.NewZonesSettingsV1(cisDomainSettingsOpt)
if cisDomainSettingsErr != nil {
	return nil, fmt.Errorf(
		"Error occured while configuring CIS Domain Settings client: %s",
		cisDomainSettingsErr)
}

return cisDomainSettingsClient, nil

```

Reply:
I have upgraded sdk and regenerated cis sdk. Modified the code as per your suggession.